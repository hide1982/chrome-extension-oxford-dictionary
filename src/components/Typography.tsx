import styled from "styled-components"
import { fontFamily } from "styles"

const Typography = styled.p`
  font-family: ${fontFamily};
  color: ${({ theme }) => theme.fontColor.main};
  font-size: 16px;
`

export default Typography
