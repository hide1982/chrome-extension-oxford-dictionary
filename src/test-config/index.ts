import "@testing-library/jest-dom"

import "@/test-config/matchMedia"
