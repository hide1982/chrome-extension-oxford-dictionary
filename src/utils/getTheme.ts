import { DARK_MODE_MEDIA_QUERY } from "@/constants"
import { lightTheme, darkTheme } from "@/styles/theme"
import type { Theme } from "@/styles/theme"

const isDarkMode = (): boolean =>
  window.matchMedia(DARK_MODE_MEDIA_QUERY).matches

const getTheme = (): Theme => {
  const theme = isDarkMode() ? darkTheme : lightTheme

  return theme
}

export default getTheme
