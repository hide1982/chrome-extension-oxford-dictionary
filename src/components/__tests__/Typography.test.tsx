import React from "react"
import "@/test-config"
import { render } from "@testing-library/react"
import { ThemeProvider } from "styled-components"

import getTheme from "@/utils/getTheme"
import Typography, { Props } from "@/components/Typography"

const setup = (props?: Props) => {
  const utils = render(
    <ThemeProvider theme={getTheme()}>
      <Typography {...props}>typography</Typography>
    </ThemeProvider>
  )
  const typography = utils.getByTestId("typography")

  return {
    typography,
    utils,
  }
}

describe("Typography component", () => {
  test("screenshot", () => {
    const { typography } = setup()

    expect(typography).toMatchSnapshot()
  })

  test("should change each styles", () => {
    const { typography } = setup({
      fontColor: "#333",
      fontSize: 12,
      fontStyle: "italic",
      fontWeight: "bold",
    })

    expect(typography).toHaveStyle({
      color: "#333",
      fontSize: "12px",
      fontStyle: "italic",
      fontWeight: "bold",
    })
  })

  test("should change font size by string", () => {
    const { typography } = setup({
      fontSize: "11px",
    })

    expect(typography).toHaveStyle({ fontSize: "11px" })
  })
})
