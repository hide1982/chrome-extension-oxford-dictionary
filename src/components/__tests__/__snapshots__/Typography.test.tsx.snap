// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component screenshot 1`] = `
<p
  class="sc-bdfBwQ fbaPVK"
  data-testid="typography"
  font-size="16"
>
  typography
</p>
`;
